certbot/plugins/pluginstorage_test.py

```diff
@@ -0,0 +1,140 @@
+"""Tests for certbot.plugins.common.PluginStorage"""
+import json
+import mock
+import os
+import shutil
+import tempfile
+import unittest
+
+from certbot import errors
+
+from certbot.plugins import common
+
+
+class PluginStorageTest(unittest.TestCase):
+    """Test for certbot.plugins.common.PluginStorage"""
+
+    def setUp(self):
+
+        class MockPlugin(common.Installer):
+            """Mock plugin"""
+            pass
+        self.plugin_cls = MockPlugin
+        self.config_dir = tempfile.mkdtemp()
+        self.config = mock.MagicMock(config_dir=self.config_dir)
+        with mock.patch("certbot.reverter.util"):
+            self.plugin = MockPlugin(config=self.config, name="mockplugin")
+
+    def tearDown(self):
+        shutil.rmtree(self.config_dir)
+
+    def test_initialize_no_configdir(self):
+        delattr(self.plugin.config, "config_dir")
+        self.plugin.storage.initialize_storage()
+        self.assertTrue(self.plugin.storage.storagepath == None)
+
+    def test_load_errors(self):
```

Comment:
nit: It might be nice to split each error here into it's own test method with a nice name. I don't care too much if you make this change or not.